### ISSUE TITLE
cannot reproduce your results of nuscene->kitti in v0.3.0

### ISSUE THREAD

**Issue by xero0519** [CVMI-Lab/ST3D#17]
hi, sorry for bother you, i cannot reproduce your results of nuscene->kitti in v0.3.0. i have noticed some one reproduce your results in v0.2.0, but i do not find the git branch to run with OpenPCDet v0.2.0. could you please help me to reproduce the performance in your paper.

**Comment by jihanyang** [comment]
Hello, you can checkout to commit b500097 to try our code in openpcdet v0.2.0. You should also re-setup the pcdet. Actually, our internal repo has been compatible for openpcdet v0.2.0 and v0.3.0, so it should not be a important factor. However, it is possible that I make some mistake when merge to pcdetv0.3.0 in this repo, and I will also reproduce it later for this repo.

**Comment by xero0519** [comment]
i am not sure that there is a problem that the evaluation performace decreased to 0.000 when apply ST3D current version with spconv-1.0.

**Comment by jihanyang** [comment]
I have test it with spconv 1.0, and this might not be the reason. Previous attempts with error pcdet and spconv version can also correctly test the checkpoint in model_zoo. Maybe you can visualize the predictions with gts.

**Comment by xero0519** [comment]
ok, i will try it right away.  thanks.

**Comment by jihanyang** [comment]
Hello,

After try to reproduce the result, I find that some unknown issues exist in applying openpcdet v0.3 for current code and obstruct reproduce the results. Therefore, I rollback the pcdet version in this repo to pcdet v0.2 again, and already reproduce the performance in nuscenes -> kitti. I apologize for my late response.

By the way, as I have claimed before, I already support pcdet v0.3 in our internal version. So, I believe that this issue is caused by my mistake when update to pcdet v0.3. I might come back to support pcdet v0.3 again.

**Comment by weijiawang96** [comment]
Sorry, I'm really having trouble with running the openpcdet v0.2.0 version on my computer, so I can only use v0.3.0 for now. Take SECOND_IOU (source only) for nuscenes -> KITTI as an example, I'll get a result that is very close to the paper, but not exactly the same. 

Here is my situation:

Command:
`python test.py --cfg_file ./cfgs/da-nuscenes-kitti_models/secondiou/secondiou_old_anchor.yaml --batch_size 4 --ckpt ./cfgs/da-nuscenes-kitti_models/secondiou/ckpt_secondiou_source_only.pth --set DATA_CONFIG_TAR.FOV_POINTS_ONLY True
`

Output:
[2021-10-04 18:27:23,801  eval_utils.py 116  INFO]  Car AP@0.70, 0.70, 0.70:
bbox AP:81.8798, 68.7139, 65.6849
bev  AP:55.8829, 45.9529, 45.2148
3d   AP:25.2459, 19.9830, 18.8851
aos  AP:81.58, 68.07, 64.86
Car AP_R40@0.70, 0.70, 0.70:
bbox AP:83.7378, 68.4617, 67.5043
bev  AP:54.1226, 43.3066, 42.3610
3d   AP:20.1308, 14.0027, 13.0358
aos  AP:83.41, 67.80, 66.51
Car AP@0.70, 0.50, 0.50:
bbox AP:81.8798, 68.7139, 65.6849
bev  AP:87.5895, 81.2958, 82.2366
3d   AP:85.5229, 74.3388, 74.6212
aos  AP:81.58, 68.07, 64.86
Car AP_R40@0.70, 0.50, 0.50:
bbox AP:83.7378, 68.4617, 67.5043
bev  AP:91.6398, 83.1709, 82.7207
3d   AP:87.8809, 75.6476, 76.0395
aos  AP:83.41, 67.80, 66.51

Can openpcdet v0.3.0 be the reason? Also I'm using spconv 1.2.1, so I don't know if it produces a different result compared to spconv 1.0. I don't know if @xero0519 is getting the same result when you use openpcdet v0.3.0.

Any suggestions are appreciated!

**Comment by jihanyang** [comment]
Hello, in the pretraining stage, openpcdet v0.2.0 and openpcdet v0.3.0 provide similar results. Nevertheless, in the self-training stage, especially pseudo label generation stage (evaluation), they give very different results. So, there must be some bugs when I merge to openpcdet v0.3.0.

**Comment by weijiawang96** [comment]
Ok, thank you so much for the details. If possible, will you consider separately releasing a branch that only supports openpcdet v0.3.0? That will avoid the issue of merging with openpcdet v0.2.0. I really appreciate your help!

**Comment by jihanyang** [comment]
Sorry, due to massive works these days, I do not have plan to support openpcdet v0.3.0 again recently.

**Comment by xero0519** [comment]
> Hello,
> 
> After try to reproduce the result, I find that some unknown issues exist in applying openpcdet v0.3 for current code and obstruct reproduce the results. Therefore, I rollback the pcdet version in this repo to pcdet v0.2 again, and already reproduce the performance in nuscenes -> kitti. I apologize for my late response.
> 
> By the way, as I have claimed before, I already support pcdet v0.3 in our internal version. So, I believe that this issue is caused by my mistake when update to pcdet v0.3. I might come back to support pcdet v0.3 again.

Yes, Thanks, i have reproduced nuscene->kitti, the performance is up to mAP=60.56 of Car_3d/moderate_R40 3D@iou0.7 by using the SN pretrained model. The environment is openpcdet v0.2.0+spconv1.0. @weijiawang96
